lib/zoom/actions.rb

```diff
@@ -26,31 +26,40 @@ def self.determine_request_options(client, oauth)
       end
     end
 
-    def self.make_request(client, method, parsed_path, params, request_options)
+    def self.make_request(args)
+      client, method, parsed_path, params, request_options, oauth =
+        args.values_at :client, :method, :parsed_path, :params, :request_options, :oauth
       case method
       when :get
         request_options[:query] = params
       when :post, :put, :patch
-        request_options[:body] = params.to_json
+        request_options[:body] =
+          oauth ? URI.encode_www_form(params.to_a) : params.to_json
       end
       client.class.public_send(method, parsed_path, **request_options)
     end
 
     [:get, :post, :patch, :put, :delete].each do |method|
       define_method(method) do |name, path, options={}|
-        required, permitted, oauth = options.values_at :require, :permit, :oauth
+        required, permitted, oauth, args_to_params, headers =
+          options.values_at :require, :permit, :oauth, :args_to_params, :headers
         required = Array(required) unless required.is_a?(Hash)
         permitted = Array(permitted) unless permitted.is_a?(Hash)
 
         define_method(name) do |*args|
           path_keys = Zoom::Actions.extract_path_keys(path)
-          params = Zoom::Params.new(Utils.extract_options!(args))
+          params = Utils.extract_options!(args)
+          args_to_params&.each { |key, value| params[value] = params.delete key if params[key] }
```

Comment:
Is the purpose of `args_to_params` to essentially allow backwards compatibility between the old API (using `auth_code`) and the correct Zoom API (using `code`)? Also `auth_token` to `token`. I might be more inclined to just update the `OAuth` class to change its calls to use the new (correct) format (`code` and `token`) and also require users to change their usage of the gem to meet the new format, with a Breaking Change noted in the Changelog. But I admit that could cause problems for many users if they don't see it. Perhaps the `OAuth` class could allow either argument as input but only use the new ones in its API calls to Zoom.

Reply:
Perhaps we keep backwards compatibility for now but add a deprecation warning and plan to remove it in the next major version change?

Just suggesting a 3rd option - your call though @joelmichael.